### ISSUE TITLE
Apps page stuck at loading

### ISSUE THREAD

**Issue by novocaiin** [ImranR98/Obtainium#378]
![Screenshot_20230321-215853](https://user-images.githubusercontent.com/95007229/226785605-442d2ca6-823f-477d-a802-47f431c641b7.png)

Been like this for weeks.
Was working beautifully before.
No idea what has changed.
All permissions are given.
Rebooting and clearing cache doesn't help.
I've searched through existing issues and couldn't find anything similar.

**Comment by ImranR98** [comment]
Could you zip up the `/Android/data/dev.imranr.obtainium` folder and attach it? It contains all your Obtainium Apps with their settings. Doesn't include App settings or GitHub credentials (if any). Also what Android version are you on?

**Comment by novocaiin** [comment]
[dev.imranr.obtainium.zip](https://github.com/ImranR98/Obtainium/files/11035802/dev.imranr.obtainium.zip)
I'm on a Pixel 5 running Graphene OS (Android 13).
I've just realized after looking through that directory that I've recently uninstalled one of the apps Obtainium was tracking (Untrackme).
I did it through Android settings.
Could that be the culprit?

**Comment by ImranR98** [comment]
> I've just realized after looking through that directory that I've recently uninstalled one of the apps Obtainium was tracking (Untrackme). I did it through Android settings. Could that be the culprit?

Hopefully not, Obtainium should detect that it was uninstalled and update the "Installed Version" shown.

**Comment by ImranR98** [comment]
Found the issue. For some reason the data file for Untrackme in Obtainium became corrupted (in this case it just got emptied completely) and the error wasn't being handled. I've added error handling in the latest release, so now corrupt data files will be marked as corrupt and ignored (but not deleted as they may be useful for further debugging).

No idea what caused the file to become corrupt though. Probably some race condition that's going to be hard to track down. I'll consider this issue closed though - you're the first to report it so hopefully it's rare enough to not be a big deal now that it doesn't crash the loading process.

https://github.com/ImranR98/Obtainium/releases/tag/v0.11.13-beta
